src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs

```diff
@@ -588,59 +593,137 @@ void EmitBindCoreCall(string varName)
-                        _ => $"{typeDisplayString}.{Identifier.Parse}({stringValueVarName})"
-                    };
+                    innerExceptionTypeDisplayString = FullyQualifiedDisplayName.FormatException;
+                    cultureInfoTypeDisplayString = FullyQualifiedDisplayName.CultureInfo;
+                    numberStylesTypeDisplayString = FullyQualifiedDisplayName.NumberStyles;
                 }
-                else if (type.SpecKind == TypeSpecKind.Enum)
+                else
                 {
-                    string enumValueVarName = GetIncrementalVarName(Identifier.enumValue);
-                    assignmentCondition += $" && {Identifier.Enum}.{Identifier.TryParse}({stringValueVarName}, true, out {typeDisplayString} {enumValueVarName})";
-                    rhs = enumValueVarName;
+                    innerExceptionTypeDisplayString = Identifier.Exception;
+                    cultureInfoTypeDisplayString = Identifier.CultureInfo;
+                    numberStylesTypeDisplayString = Identifier.NumberStyles;
                 }
-                else if (type.SpecKind == TypeSpecKind.ByteArray)
+
+                _writer.WriteBlockStart($"if ({expressionForConfigStringValue} is string {stringValueVarName})");
+
+                if (typeKind is StringParsableTypeKind.ConfigValue)
                 {
-                    rhs = $"{Expression.ConvertFromBase64String}({stringValueVarName})";
+                    EmitAssignment(expressionForMemberAccess, stringValueVarName);
+                    writeOnSuccess?.Invoke();
+                    _writer.WriteBlockEnd();
+                    return;
                 }
-                else
+                else if (typeKind is StringParsableTypeKind.Uri)
                 {
+                    string uriVarName = GetIncrementalVarName(Identifier.temp);
+                    _writer.WriteLine($"{Identifier.Uri}.{Identifier.TryCreate}({stringValueVarName}, {Identifier.UriKind}.{Identifier.RelativeOrAbsolute}, out {Identifier.Uri}? {uriVarName});");
+                    _writer.WriteBlock($$"""
+                        if ({{uriVarName}} is not null)
+                        {
+                            {{expressionForMemberAccess}} = {{uriVarName}};
+                        }
+                        """);
+                    _writer.WriteBlockEnd();
                     return;
                 }
 
-                _writer.WriteBlockStart($"if ({assignmentCondition})");
-                EmitAssignment(expressionForMemberAccess, rhs);
-                writeExtraOnSuccess?.Invoke();
+                // Types we catch exceptions for.
+                string rhs;
+                switch (typeKind)
+                {
+                    case StringParsableTypeKind.Enum:
+                        {
+                            rhs = $"({typeDisplayString}){Identifier.Enum}.{Identifier.Parse}(typeof({typeDisplayString}), {stringValueVarName}, true)";
```

Comment:
Please label parameters that are passed literals for easier readability:
```suggestion
                            rhs = $"({typeDisplayString}){Identifier.Enum}.{Identifier.Parse}(typeof({typeDisplayString}), {stringValueVarName}, ignoreCase: true)";
```